FilmFestivalLoader/IDFA/parse_idfa_html.py

```diff
@@ -773,515 +437,598 @@ class FilmsParseState(Enum):
-                debug_text = '\n'.join([
-                    error_desc,
-                    f'{"registered url":-<20}{self.film.url}',
-                    f'{"corrected url":-<20}{self.corrected_url}',
-                ])
-                ERROR_COLLECTOR.add('DIFFERENT TITLE', error_desc)
-                self.print_debug(f'DIFFERENT TITLE: {debug_text}')
-            self.state_stack.change(self.FilmParseState.AWAITING_META_DICT)
-        elif self.state_stack.state_is(self.FilmParseState.IN_META_PROPERTY):
-            self.film_property_by_label[self.metadata_key] = data
-        elif self.state_stack.state_is(self.FilmParseState.IN_PAGE_SECTION):
-            if data == 'Synopsis':
-                self.state_stack.change(self.FilmParseState.AWAITING_ARTICLE)
-            elif data.startswith('Tickets'):
-                self.state_stack.change(self.FilmParseState.AWAITING_SCREENINGS)
-            elif data == 'Credits':
-                self.state_stack.change(self.FilmParseState.AWAITING_CREDITS)
-            elif data == 'Stills':
-                self.update_film_info()
-                self.state_stack.change(self.FilmParseState.DONE)
-        elif self.state_stack.state_is(self.FilmParseState.IN_SCREENING_INFO):
-            self.process_screening_info(data)
-        elif self.state_stack.state_is(self.FilmParseState.IN_ARTICLE):
-            if not data.startswith('.css'):
-                self.article_paragraph += data
-        elif self.state_stack.state_is(self.FilmParseState.IN_SCREENING_DATE):
-            if self.set_screening_date(data):
-                self.state_stack.change(self.FilmParseState.AWAITING_SCREENING_INFO)
-            else:
-                self.state_stack.pop()
-        elif self.state_stack.state_is(self.FilmParseState.IN_TIMES):
-            if self.set_screening_times(data):
-                self.state_stack.change(self.FilmParseState.AWAITING_LOCATION)
-            else:
-                self.state_stack.pop()
-        elif self.state_stack.state_is(self.FilmParseState.IN_LOCATION):
-            self.screen = self.get_idfa_screen(data)
-            self.add_idfa_screening(True)
-            self.state_stack.pop()
+        stack = self.state_stack
+        state = self.PathwayFilmPageParseState
+        match stack.state():
+            case state.IN_TITLE:
+                self.film_title = data.strip()
+            case state.IN_PARAGRAPH:
+                self.add_article_text(data)
+            case state.IN_DATE:
+                self._set_screening_date(data)
+            case state.IN_TIME:
+                self._set_screening_times(data)
+            case state.IN_LOCATION:
+                if self._set_idfa_screen(data):
+                    self._add_screening()
+                    DEBUG_RECORDER.add(f'SCREEN: {data=}, {self.screen=}')
+                    stack.pop(2)  # Pop stack twice to get back to "AWAITING_SCREENINGS".
+            case state.IN_SUB_HEADER:
```

Comment:
Please consider to create a function to do this, returning whether the stack can be popped.

Reply:
Already done in the context of a comment above.